src/binaryen-shell.cpp

```diff
@@ -275,20 +272,22 @@ static void run_asserts(size_t* i, bool* checked, AllocatingModule* wasm,
     if (id == ASSERT_INVALID) {
       // a module invalidity test
       AllocatingModule wasm;
-      bool invalid = false;
-      jmp_buf trapState;
-      if (setjmp(trapState) == 0) {
-        *builder = std::unique_ptr<SExpressionWasmBuilder>(new SExpressionWasmBuilder(wasm, *curr[1], [&]() {
-          invalid = true;
-          longjmp(trapState, 1);
-        }));
-      }
       if (print_before || print_after) {
         Colors::bold(std::cout);
         std::cerr << "printing in module invalidity test:\n";
         Colors::normal(std::cout);
         std::cout << wasm;
       }
+      bool invalid = false;
+      try {
+        *builder = std::unique_ptr<SExpressionWasmBuilder>(
+            new SExpressionWasmBuilder(wasm, *curr[1], [&]() {
+              invalid = true;
+              throw TrapException();
```

Comment:
hmm actually this and the catch below should be for a third type, a `ParseException` perhaps? It's not a runtime trap.